File: 038__Count_And_Say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        def recursive(n):
            result = ""
            count = 1
            if len(n) == 1:
                return "1" + n
            for i in range(1, len(n)):
                if n[i] == n[i - 1]:
                    count += 1
                else:
                    result += str(count) + n[i - 1]
                    count = 1
            result += str(count) + n[-1]
            return result

        result = "1"
        while int(n) - 1 :
            print(result)
            result = recursive(result)
            n -= 1
        return str(result)

File: Java/038__Count_and_Say.java
class Solution {
    public String countAndSay(int n) {
        String s = "1";
        while (n > 1) {
            s = helper(s);
            n--;
        }
        return s;
    }

    private String helper(String s) {
        int count = 1;
        String result = "";
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == s.charAt(i - 1)) {
                count++;
            } else {
                result += String.valueOf(count) + Character.toString(s.charAt(i - 1));
                count = 1;
            }
        }
        result += String.valueOf(count) + Character.toString(s.charAt(s.length() - 1));
        return result;
    }
}
